syntax = "proto2";

import "primitives.proto";

package troll;

message Scene {
  optional string id = 1;
  optional string scene_manager = 2;
  optional Box viewport = 3;

  optional BitmapSceneManagerConfig bitmap_config = 4;

  repeated InitScript on_init = 5;
}

message BitmapSceneManagerConfig {
  optional string bitmap = 1;
  optional RGBa background_colour = 2;

  optional int32 width = 3;
  optional int32 height = 4;
}

message InitScript {
  optional string module = 1;
  optional string function = 2;
}
